### DISCUSSION TITLE
Retrieve more data from point picking

### DISCUSSION THREAD

**Discussion by matudinglin** [pyvista/pyvista#4628] [Q&A] [answered]
Hi everyone,

I have a question regarding the `enable_point_picking` function. Currently, it only saves the position of the picked point in the variable `picked_point`. However, I was wondering if there is a way to retrieve more information from the picked point, such as its labels or additional data. Is there a method or approach to do that?

Any insights or suggestions would be greatly appreciated.

Thank you!

**Comment by darikg** [comment, marked as answer]
Use the `use_picker` kwarg, added in pyvista 0.40:

```python
from pyvista import Sphere, Plotter
from numpy.random import uniform

sphere = Sphere()
sphere.point_data['custom_data'] = uniform(0, 1, sphere.n_points)

def my_callback(_point, picker):
    # picker is a vtkPointPicker
    point_idx = picker.GetPointId()
    val = sphere.point_data['custom_data'][point_idx]
    print(f"Clicked point {point_idx} with value {val}")

p = Plotter()
p.add_mesh(sphere, pickable=True)
p.enable_point_picking(callback=my_callback, use_picker=True)
p.show()
```

**Reply by matudinglin** [reply]
Hi @darikg, Thanks for your reply! I haven't updated pyvista to 0.40, let me check it out.

**Reply by darikg** [reply]
no prob! before 0.40 it should be `use_mesh` instead of `use_picker`, and 

```python
def my_callback(mesh, point_idx):
    val = mesh.point_data['custom_data'][point_idx]
    print(f"Clicked on point {point_idx} with custom data {val}")
```

**Reply by dkobayas-cyber** [reply]
@darikg Thank you for the excellent feature. I am trying to run your code on my jupyter notebook with `left_clicking` enabled, but the point picking does not work. Do you have any idea what is causing this issue? My PyVista version is 0.41.1.

**Reply by darikg** [reply]
@dkobayas-cyber could you please open a new issue? what do you mean doesn't work?

**Reply by dkobayas-cyber** [reply]
@darikg The problem has been fixed after updating my conda version. Until yesterday, point picking didn't show the text information ("Clicked point..."). Thank you for your resposnse anyway!
